**Question 1**: Given some sample data, write a program to answer the following: click here to access the required data set

On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 

Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 
What metric would you report for this dataset?
What is its value?


1-Importing Data and Create a Data Frame

In [2]:
import numpy as np
import pandas as pd 
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/shpify3/data3.csv


In [3]:
df=pd.read_csv('/kaggle/input/shpify3/data3.csv')


In [4]:
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

        created_at  
0  3/13/2017 12:36  
1   3/3/2017 17:38  
2   3/14/2017 4:23  
3  3/26/2017 12:43  
4    3/1/2017 4:35

2- Analysis of Data 

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   order_id        5000 non-null   int64 
 1   shop_id         5000 non-null   int64 
 2   user_id         5000 non-null   int64 
 3   order_amount    5000 non-null   int64 
 4   total_items     5000 non-null   int64 
 5   payment_method  5000 non-null   object
 6   created_at      5000 non-null   object
dtypes: int64(5), object(2)
memory usage: 273.6+ KB


In [6]:
df.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

In [9]:
#Percentage of orders more than 1000
np.sum(df['order_amount'] > 1000)/5000*100

1.4200000000000002

" Order amount" has a huge amount of standard deviation and 75% of the orders have values smaller than 390.Only 71 orders from 5000(less than 2%)  orders have value more than 1000.  So, the mean value(3145.128) is not a good representative of AOV. The Median is a better metric to describe the order amount status.

In [8]:
#Calculate Median
df['order_amount'].median()

284.0

Mode is also in a range of around 200.



In [10]:
#Calculate Mode
df['order_amount'].mode()

0    153
dtype: int64

Summary: Only one statical metric like MEAN is not enough to give a great overall view of data patterns. It is better to calculate Median and Mode of data and select one that is the best representative of majority of Order values.  I believe Median is the best metric for the data . I calculated it in my program.